src/audio/igo_nr/igo_nr.c

```diff
@@ -69,179 +69,316 @@ static void igo_nr_lib_process(struct comp_data *cd)
 	}
 }
 
+static void *igo_cir_wrap(void *ptr, void *cir_buf_end, size_t cir_buf_size)
+{
+	if (ptr > cir_buf_end)
+		ptr = (char *)ptr - cir_buf_size;
+
+	return ptr;
+}
```

Comment:
static inline void *cir_buf_wrap(void *ptr, void *buf_addr, void *buf_end)
{
	if (ptr >= buf_end)
		ptr = (char *)buf_addr +
			((char *)ptr - (char *)buf_end);

	assert((intptr_t)ptr <= (intptr_t)buf_end);

	return ptr;
}
 why not directly use this?

Reply:
I don't see issues in what I proposed.

Reply:
check https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/audio_stream.h#L414
you don't need define a new one.

Reply:
OK, I've changed and tested it now - pushing new version!